doc/sphinx/dev_git_intro.rst

```diff
@@ -52,46 +52,211 @@ Clone a local repository onto your machine
 .. (TODO: `pip install -v .`, other installation instructions...)
 
 Start coding
-^^^^^^^^^^^^
+------------
 
-- Switch to the ``develop`` branch: ``git checkout develop``.
-- If it's been a while since you created your fork, other people may have updated NOAA's ``develop`` branch. To make sure you're up-to-date, get these changes with ``git pull upstream develop`` and ``git submodule update --recursive --remote``.
-- That command updates the working copy on your computer, but you also need to tell your fork on GitHub about the changes: ``git push origin develop``.
-- Now you're up-to-date and ready to start working on a new feature. ``git checkout -b feature/<my_feature_name>`` will create a new branch (``-b`` flag) off of ``develop`` and switch you to working on that branch.
+1. Switch to the ``develop`` branch
+::
+  git checkout develop
+
+2. Make sure that you pull in changes from the develop branch frequently to simplify the merge process after you submit a PR. Update your local copy (the copy on your computer) of the develop branch
+::
+  git fetch upstream develop
+  git pull upstream develop
+  git submodule update --recursive --remote
+
+3. Next, update your remote copy (the branch on your Github fork)
+::
+  git push origin develop
+
+4. Now your branch is up-to-date, and you are ready to start working on a new feature
+::
+  git checkout -b feature/<my_feature_name>
+
+will create a new branch (``-b`` flag) off of ``develop`` and switch you to working on that branch.
+
+Updating your feature branch by merging in changes from the develop branch
+---------------------------------------------------------------------------
+1. Update the local and remote develop branches on your fork as described steps 1--3  of the **Start Coding** section, check out your feature branch, and merge the develop branch into your feature branch
```

Comment:
Do I understand correctly that you recommend updating our local (& remote) develop branch with changes from upstream develop, and then using that to merge changes into our feature branch, instead of simply pulling upstream develop from the local feature, and pushing that to origin/develop? To me the latter seems simpler, so please advise!

Reply:
Also, I assume we would push merges from our feature branch to origin/develop in order to to the pull request from there? Or is it best practice to keep our origin/develop identical to upstream/develop-- so the changes from our new feature go to upstream/develop and then we pull them into origin/developo from there?

Reply:
1. Yes, exactly. The remote develop branch on your fork (i.e., the branch on the github server) and your local develop branch (the one on your machine) will both be updated. You then sync the local feature branch with the local develop branch, then continue working on your feature branch.  `git push -u origin feature_branch` will update the remote copy of your feature branch by default. You submit PRs of the feature branch to develop

2. I answered this in another response, but will clarify that you shouldn't merge your working branch into the remote develop branch. It is best practice to leave the local copy of the develop branch alone, too, since the method I've describe does `fetch+pull` to merge changes from the remote develop branch to the local develop branch before updating the feature branch. I'll add some info on doing a "hard reset"  in case you make changes on a local branch that you don't want to keep.